doc: matter: Update Matter certification page (#12642)
nrfconnect/sdk-nrf: doc/nrf/protocols/matter/end_product/certification.rst

   :alt: Matter certification by similarity overview

   Matter certification by similarity overview

.. _ug_matter_device_certification_matter_samples:

Certification status of Matter samples
**************************************

The |NCS| includes several :ref:`matter_samples` that are example implementations of Matter devices.

> The samples are maintained to fulfill Matter certification requirements as much as possible, however they did not get official Matter compliance certificates.

Reviewer: ```suggestion
The samples are maintained to fulfill Matter certification requirements as closely as possible, but they do not have official Matter compliance certificates.
```

> You can use them as a reference for creating an own application, however it does not guarantee that the application will pass the Matter certification.

Reviewer: ```suggestion
You can use them as a reference for creating your own application, but this does not guarantee that your application will pass the Matter certification.
```

The Matter certification program currently does not provide any form of platform certification, so the product maker is solely reponsible for ensuring that the application used fulfills all Matter certification requirements.

> Before you start applying for the Matter certification, you should make sure that the application configuration generated using :file:`.zap` file is compatible with the PICS generated for the certification purpose.

Reviewer: ```suggestion
Before you start the application process for the Matter certification, make sure that the application configuration generated using the :file:`.zap` file is compatible with the PICS generated for the certification purpose.
```

> You can find the information about the specification and how to select PICS for you product on the `CSA Matter Resource Kit`_ page, in the `Specification` section.

Reviewer 2: there is no information how to select PICS but under Specification the valid PICS version is shared (which need to be used for testing/certifcation)

> Especially, make sure that:

Reviewer 2: I will even inform to check data model for each cluster, to verify 'global attributes' so:
Feature Map
Cluster Revision
Attribute List
Accepted Command List
Generated Command List
Event List (this is currently not supported even for future Matter 1.2)

Author: Added 3rd bullet point including all the list attributes.

> * ``FeatureMap`` attributes for all clusters match the desired configuration selected in PICS and all other cluster requirements related to the specific ``FeatureMap`` value are met.

Reviewer: ```suggestion
* ``FeatureMap`` attributes for all clusters match the desired configuration selected in PICS, and all other cluster requirements related to the specific ``FeatureMap`` value are met.
```

> * ``ClusterRevision`` attributes for all clusters match the value of Matter stack version that you want to use for you application certification.

Reviewer: ```suggestion
* ``ClusterRevision`` attributes for all clusters match the value of the Matter stack version that you want to use for your application certification.
```

> To read about how to configure your application using the ZAP Tool, see `ug_matter_creating_accessory`_ page.

Reviewer: ```suggestion
See the `ug_matter_creating_accessory`_ page for how to configure your application using the ZAP Tool.
```

> Before you request the official product `ug_matter_device_certification_testing`_ from ATL, it is recommended to perform the in-house product testing.

Reviewer: ```suggestion
Before you request the official product `ug_matter_device_certification_testing`_ from ATL, it is recommended to perform in-house certification testing of the product.
```

> The initial verification allows to spot majority of problems that could lead to further certification failure, and in result safe your time and money spend on official certification in ATL laboratory.

Reviewer: ```suggestion
This testing allows you to avoid most problems that could lead to further certification failure, meaning it saves time and money that would be spent on official certification in ATL laboratory.
```

Author: I'm not sure whether we should use "avoid" word. What I wanted to write is that the intention of testing in-house is to detect firmware bug earlier and be able to fix it before spending money on testing in ATL.

Reviewer: In that case, something like this is probably better:
```suggestion
This testing can help you detect problems that could lead to certification failure, meaning it saves time and money that would be spent on official certification in ATL laboratory.
```

> You can find the information about how to perform in-hourse testing using the Matter Test Harness on the `CSA Matter Resource Kit`_ page, in the `Testing` section.

Reviewer 2: ```suggestion
You can find the instruction how to execute certification tests using Matter Test Harness in-house on the `CSA Matter Resource Kit`_ page, in the `Testing` section under `Test Tool User Guide` link.
```

> You can find the instruction how to execute certification tests using Matter Test Harness in-house on the `CSA Matter Resource Kit`_ page, in the `Testing` section under `Test Tool User Guide` link.

Reviewer: ```suggestion
You can find the instruction how to execute in-house certification tests using a Matter Test Harness on the `CSA Matter Resource Kit`_ page.
Open the **Test Tool User Guide** link in the **Testing** section under the **Testing and Certification** section of the page.
```
Unfortunately, we don't have a formatting for referring to

You can use them as a reference for creating your own application, but this does not guarantee that your application will pass the Matter certification.

The Matter certification program currently does not provide any form of platform certification, so the product maker is solely reponsible for ensuring that the application used fulfills all Matter certification requirements.
Before you start the application process for the Matter certification, make sure that the application configuration generated using :file:`.zap` file is compatible with the PICS generated for the certification purpose.
You can find the information about the specification and the valid PICS for specific Matter stack version that should be used for your product on the `CSA Matter Resource Kit`_ page, in the `Specification` section.
Especially, make sure that:

* ``FeatureMap`` attributes for all clusters match the desired configuration selected in PICS, and all other cluster requirements related to the specific ``FeatureMap`` value are met.
* ``ClusterRevision`` attributes for all clusters match the value of the Matter stack version that you want to use for your application certification.

> * ``Attribute List``, ``Accepted Command List`` and ``Generated Command List`` attributes for all clusters contain all the values that were meant to be there and nothing more.

Reviewer: ```suggestion
* ``Attribute List``, ``Accepted Command List`` and ``Generated Command List`` attributes for all clusters contain all the values that need to be there and nothing more.
```
"Were meant to be" sounds kind of like a fairy tale.
